src/viem/saga.ts

```diff
@@ -200,7 +287,7 @@ function* getTransferSimulateContract({
   })
 
   return () =>
-    publicClient.celo.simulateContract({
+    publicClient[network].simulateContract({
```

Comment:
This should in theory "just work" for non-Celo ERC20 tokens (i.e., those with an address). I haven't been able to test this though, since my test wallet only has ETH.